testschematic/tests.go

```diff
@@ -142,46 +142,71 @@ func (options *TestSchematicOptions) RunSchematicTest() error {
 			}
 		}
 	}
-
-	// ------ DESTROY ------
-	// only run destroy if we had potentially created resources
-	if svc.TerraformResourcesCreated {
-		// Check if "DO_NOT_DESTROY_ON_FAILURE" is set
-		envVal, _ := os.LookupEnv("DO_NOT_DESTROY_ON_FAILURE")
-		if options.Testing.Failed() && strings.ToLower(envVal) == "true" {
-			options.Testing.Log("[SCHEMATICS] Schematics APPLY failed. Debug the Test and delete resources manually.")
-		} else {
-			destroyResponse, destroyErr := svc.CreateDestroyJob()
-			if assert.NoErrorf(options.Testing, destroyErr, "error creating DESTROY - %s", workspaceNameString) {
-				options.Testing.Log("[SCHEMATICS] Starting DESTROY job ...")
-				destroyJobStatus, destroyStatusErr := svc.WaitForFinalJobStatus(*destroyResponse.Activityid)
-				if assert.NoErrorf(options.Testing, destroyStatusErr, "error waiting for DESTROY to finish - %s", workspaceNameString) {
-					assert.Equalf(options.Testing, SchematicsJobStatusCompleted, destroyJobStatus, "DESTROY has failed with status %s - %s", destroyJobStatus, workspaceNameString)
-				}
-			}
-		}
-	}
+	options.SchematicsTestSvc = svc
+	options.testTearDownResources()
 
 	return nil
 }
 
+// Function to destroy all resources. Resources are not destroyed if tests failed and "DO_NOT_DESTROY_ON_FAILURE" environment variable is true.
+// If options.ImplicitDestroy is set then these resources from the State file are removed to allow implicit destroy.
+func (options *TestSchematicOptions) TestTearDown() {
+	options.SkipTestTearDown = false
+	options.testTearDownResources()
+	defer testTearDownWorkspace(options)
+}
+
 // testTearDown is a helper function, typically called via golang "defer", that will clean up and remove any existing resources that were
 // created for the test.
 // The removal of some resources may be influenced by certain conditions or optional settings.
-func testTearDown(svc *SchematicsTestService, options *TestSchematicOptions) {
+func (options *TestSchematicOptions) testTearDownResources() {
```

Comment:
If tear down of resources is private function (cannot be called outside of this package), then it shouldn't be a method on "options" it should be a method on SchematicsTestService.